opentracing-mock/src/main/java/io/opentracing/mock/MockSpan.java

```diff
@@ -94,8 +95,11 @@ public void finish() {
 
     @Override
     public synchronized void finish(long finishMicros) {
-        this.finishMicros = finishMicros;
-        this.mockTracer.appendFinishedSpan(this);
+        if (!finished) {
```

Comment:
I agree. In fact, based on "mainly used for testing" argument, it's not unreasonable to even throw an exception (or at least make it a configuration option). It is usually a sign of poor instrumentation, not a problem that needs fixing in the tracer.